src/Microsoft.HttpRepl.IntegrationTests/Commands/SetSwaggerCommandTests.cs

```diff
@@ -0,0 +1,324 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.HttpRepl.Commands;
+using Microsoft.HttpRepl.IntegrationTests.Mocks;
+using Microsoft.Repl;
+using Microsoft.Repl.ConsoleHandling;
+using Microsoft.Repl.Parsing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.HttpRepl.IntegrationTests.Commands
+{
+    public class SetSwaggerCommandTests
+    {
+        [Fact]
+        public async Task ExecuteAsync_WithExactlyOneParseResultSection_WritesToConsoleManagerError()
```

Comment:
I've been going back and forth about whether we really should be testing for this. For the end user, there's no way for them to execute the SwaggerCommand without having at least two sections in the command line - "set" and "swagger". And this test wouldn't tell us if someone made a shortcut (set-swagger?) elsewhere in the code and changed ExecuteAsync to look for it and expected it to run, because this is only testing for one section with the text "section1". 
I don't know, something feels off. Thoughts?

Reply:
I'm testing this line of code: https://github.com/aspnet/HttpRepl/blob/dev/bhsubra/SetSwaggerCommandTests/src/Microsoft.HttpRepl/Commands/SetSwaggerCommand.cs#L246

If someone made a shortcut elsewhere in code and changed ExecuteAsync to look for it, this test needs to be updated to accommodate that change, no? Let me know if I am missing something.

Reply:
Hmmm, now when I think of it, the error checks in ExecuteAsync(..): https://github.com/aspnet/HttpRepl/blob/dev/bhsubra/SetSwaggerCommandTests/src/Microsoft.HttpRepl/Commands/SetSwaggerCommand.cs#L246 are probably meaningless. CanHandle(..) will fail if set swagger is not present. So it won't hit ExecuteAsync(..). The check for ValidUri seems ok. We probably don't need parseResult.Sections.Count != 3 || string.IsNullOrEmpty(parseResult.Sections[2] check at all.

Reply:
I think you're now having the same back and forth I was :) Your second comment is what I was thinking... those checks shouldn't ever fail in the (current) normal flow of code. In some future change I guess there could be some other way of executing the functions that didn't call that CanHandle check first?

Reply:
This is the current behavior:
1. Executing - set swagger does nothing. ExecuteAsync(..) gets called but since uri is not provided, it just sets the HttpState SwaggerStructure to null
2. Specifying invalid uri - set swagger test, writes an error to console. Below is complete output from console:
https://localhost:44369/~ set swagger test
Must specify a swagger document
3. When you specify sections greater than 3, says- set swagger https://localhost:44369/swagger/v1/swagger.json test2, has same console output as 2

I started working on tests for UI command and I see same weirdness in error reporting there as well. I created a new work item in azure devops and tagged Ahmed(Can't tag him here for some reason) for his input. Let me know if anyone has any concerns or anything to add here. Thanks!